class.jetpack-client.php

```diff
@@ -285,4 +300,36 @@ static function wpcom_json_api_request_as_blog( $path, $version = self::WPCOM_JS
 		return Jetpack_Client::remote_request( $validated_args, $body );
 	}
 
+	/**
+	 * Takes an array or similar structure and recursively turns all values into strings. This is used to
+	 * make sure that body hashes are made ith the string version, which is what will be seen after a
+	 * server pulls up the data in the $_POST array.
+	 *
+	 * @param array|mixed $data
+	 *
+	 * @return array|string
+	 */
+	public static function _stringify_data( $data ) {
```

Comment:
But if using PHP's serialize and unserialize, the type is also maintained -- as an integer, string, whatever, rather than turning everything into a string.